unittests/CAS/InMemoryCASDatabaseTests.cpp

```diff
@@ -0,0 +1,38 @@
+//===- InMemoryCASDatabaseTests.cpp ---------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "llbuild/CAS/InMemoryCASDatabase.h"
+
+#include "gtest/gtest.h"
+
+using namespace llbuild;
+using namespace llbuild::CAS;
+
+TEST(InMemoryCASDatabaseTests, basic) {
+  InMemoryCASDatabase db;
+
+  // Put an trivial object.
+  auto id = db.put("X").get().get();
```

Comment:
`.get().get()`...    sadness.   Makes sense, but not particularly friendly/ergonomic.

Don't know llvm::ErrorOr... what happens if it actually contains an error?